### PR TITLE
Emit `OpModfStruct` for GLSL `EOpModf` instead of `OpModf` 

### LINKED ISSUE #3756: OpModf is deprecated, should emit OpModfStruct instead
As per https://registry.khronos.org/SPIR-V/specs/unified1/GLSL.std.450.html#_changes_from_version_1_00_revision_11, `OpModf` is deprecated and `modf` should be emitted as `OpModfStruct` as is done for `frexp` currently.

### PR DESCRIPTION
**Pull request by chyyran**
My attempt at fixing https://github.com/KhronosGroup/glslang/issues/3756.

Also updated the tests to account for the new type getting added.

### CODE REVIEW

**Comment by chyyran** [comment]
Failing unit tests

```
2: [  FAILED  ] Glsl/CompileVulkanToSpirvTest.FromFile/spv_400_frag, where GetParam() = "spv.400.frag"
2: [  FAILED  ] Glsl/CompileVulkanToSpirvTest.FromFile/spv_Operations_frag, where GetParam() = "spv.Operations.frag"
2: [  FAILED  ] Glsl/CompileVulkan1_1ToSpirvTest.FromFile/spv_float32_frag, where GetParam() = "spv.float32.frag"
2: [  FAILED  ] Glsl/CompileVulkan1_1ToSpirvTest.FromFile/spv_float64_frag, where GetParam() = "spv.float64.frag"
2: [  FAILED  ] Glsl/CompileVulkanToSpirvTestAMD.FromFile/spv_float16_frag, where GetParam() = "spv.float16.frag"
```

**Comment by chyyran** [comment]
Seems like more tests need to be updated. I ran glslangtests with --update-mode but it seems more needs to be done?

**Comment by chyyran** [comment]
Managed to get the `runtests` tests to run and updated those results as well.

**Comment by chyyran** [comment]
@arcady-lunarg any updates on getting this merged?

File: SPIRV/GlslangToSpv.cpp [hunk 1]
```diff
@@ -8805,7 +8805,17 @@ spv::Id TGlslangToSpvTraverser::createMiscOperation(glslang::TOperator op, spv::
         builder.promoteScalar(precision, operands.front(), operands.back());
         break;
     case glslang::EOpModf:
-        libCall = spv::GLSLstd450Modf;
+        {
+            libCall = spv::GLSLstd450ModfStruct;
+            assert(builder.isFloatType(builder.getScalarTypeId(typeId0)));
+            int width = builder.getScalarTypeWidth(typeId0);
+            if (width == 16)
+                // Using 16-bit whole number operand, enable extension E_SPV_AMD_gpu_shader_half_float
```
Comment (arcady-lunarg) [inline SPIRV/GlslangToSpv.cpp:8813]: I think this can be removed, the code is kind of self-explanatory

File: SPIRV/GlslangToSpv.cpp [hunk 2]
```diff
@@ -8805,7 +8805,17 @@ spv::Id TGlslangToSpvTraverser::createMiscOperation(glslang::TOperator op, spv::
...
+                builder.addExtension(spv::E_SPV_AMD_gpu_shader_half_float);
+            // The two members of the returned struct and x must all be the same type.
```
Comment (arcady-lunarg) [inline SPIRV/GlslangToSpv.cpp:8815]: This comment should maybe be more like "The returned struct has two members of the same type as the first argument"

**Review by arcady-lunarg — CHANGES_REQUESTED** [review]
The code actually looks fine, I don't really like the way the comments are worded though, could you please reword them?

**Comment by chyyran** [comment]
@arcady-lunarg address nits as of 5c784d6

**Review by arcady-lunarg — APPROVED** [review]

**Pull request merged**